### PR TITLE
use single info point that is extendable

### PR DESCRIPTION
**Pull request by moovida**
Hi @tuarrep , thanks for this amazing plugin, I love it :-)

In this pull request I would like to make a proposal to use hoverpoints that:

* extend LatLng so that the point list can be used both in the polyline and in the Elevation
* change the ElevationHoverNotification to a InfoHoverNotification, and pass that as hover notification instead of only the coordinate. That would allow to extend that InfoPoint and pass through it also other information like progressive distance, speed and timestamp.

I hope you like the idea.

### CODE REVIEW

**Comment by tuarrep** [comment]
Thank you for this PR!

I need to review it deeper later, but here some first impressions. 

I'm convinced by the `InfoPoint` class. It's smarter than mine.

I'm not convinced by the name. "Info" is too vague. What about `ElevatedLatLng`?

I don't understand the benefits of the changes on the notification. If I've well understood, it's just a name change.

BTW if it's really needed, it should be a second PR to split responsibilities and allow reverting if needed.

Thanks again for your interest.

**Comment by moovida** [comment]
> I'm convinced by the InfoPoint class. It's smarter than mine.

ok, that is great.

> I'm not convinced by the name. "Info" is too vague. What about ElevatedLatLng?

I chose info (even if I am also not really happy about it), because it can carry anything. But then you are right. In fact downstream apps will anyways extend the ElevatedLatLng if they add more information. 
I agree, most probably then your original name would be best to keep?

> I don't understand the benefits of the changes on the notification. If I've well understood, it's just a name change.

Yes, I changed the name, and after the discussion above, I think it should stay as it was.
The important thing is that the notification carries the complete point and not only the LatLng information.

I could redo the PR reflecting this discussion. What do you think?

**Comment by moovida** [comment]
Closing this pull request and opening a new one.

**Comment by tuarrep** [comment]
Just let this one open for the new `ElevationPoint`, remove the notification renaming

**Comment by tuarrep** [comment]
To prevent BC we can leave the point named `ElevationPoint`.

File: lib/map_elevation.dart [hunk 1]
```diff
@@ -246,11 +244,11 @@ class _ElevationPainter extends CustomPainter {
 }
 
 /// [Notification] emitted when graph is hovered
-class ElevationHoverNotification extends Notification {
-  /// Hovered point coordinates
-  final lg.LatLng position;
+class InfoHoverNotification extends Notification {
```
Comment (tuarrep) [inline lib/map_elevation.dart:247]: Do not rename

File: lib/map_elevation.dart [hunk 2]
```diff
@@ -269,12 +267,10 @@ class ElevationGradientColors {
 }
 
 /// Geographic point with elevation
-class ElevationPoint {
-  /// Latitude and Longitude
-  lg.LatLng latLng;
-
+class InfoPoint extends lg.LatLng {
```
Comment (tuarrep) [inline lib/map_elevation.dart:270]: Do not rename this to avoid BC

File: lib/map_elevation.dart [hunk 3]
```diff
@@ -269,12 +267,10 @@ class ElevationGradientColors {
...
   /// Altitude (in meters)
   double altitude;
 
-  ElevationPoint(this.latLng, this.altitude);
+  InfoPoint(double latitude, double longitude, this.altitude)
+      : super(latitude, longitude);
 }
```
Comment (tuarrep) [inline lib/map_elevation.dart:276]: Add a getter `latLng` which return the current instance for backward compatibility
Add a doc comment that deprecates it

**Review by tuarrep — COMMENTED** [review]
Made some comments to avoid BC

**Comment by moovida** [comment]
@tuarrep I had made the new pull request in order to not revert changes in this one (I am not sure that it would be so readable). Would you mind give a quick check to the other PR? I think it reflects almost everything you commented. 

I would have to add as per your comments: 

* Add a getter latLng which return the current instance for backward compatibility
* Add a doc comment that deprecates it

**Comment by tuarrep** [comment]
OK.

Closing in favor of #2

**Pull request closed without merging**